apps/functions-subscription/src/adapters/azure/functions/list-subscriptions-by-traial-id.ts

```diff
@@ -0,0 +1,83 @@
+import * as H from '@pagopa/handler-kit';
+import { pipe, flow } from 'fp-ts/lib/function';
+import * as RTE from 'fp-ts/ReaderTaskEither';
+import * as t from 'io-ts';
+import { httpAzureFunction } from '@pagopa/handler-kit-azure-func';
+import { Trial as TrialAPI } from '../../../generated/definitions/internal/Trial';
+import { SystemEnv } from '../../../system-env';
+import { getAndValidateUser, parseQueryParameter, parsePathParameter } from './middleware';
+import { toHttpProblemJson } from './errors';
+import { toTrialAPI } from './codec';
+import { TrialIdCodec } from '../../../domain/trial';
+import {
+  IWithinRangeIntegerTag,
+  NumberFromString,
+  WithinRangeInteger,
+} from '@pagopa/ts-commons/lib/numbers';
+import { withDefault } from '@pagopa/ts-commons/lib/types';
+
+type Env = Pick<SystemEnv, 'getSubscriptionByTrialId'>;
```

Comment:
**question**: same for a previous comment.

here the function is called `getSubscriptionByTrialId`, but the query looks like we are listing subscriptions